cesiumTestsExtension/public/CesiumTests.cpp

```diff
@@ -0,0 +1,93 @@
+#define CARB_EXPORTS
+#define DOCTEST_CONFIG_IMPLEMENT
+#define DOCTEST_CONFIG_IMPLEMENTATION_IN_DLL
+#define DOCTEST_CONFIG_SUPER_FAST_ASSERTS
+
+#include "CesiumTests.h"
+
+#include "cesium/omniverse/Context.h"
+
+#include <carb/PluginUtils.h>
+#include <doctest/doctest.h>
+#include <omni/fabric/IFabric.h>
+
+#include <iostream>
+
+static void exampleTest() {
+    // This file is set up such that we can use doctest assertions (e.x. CHECK)
+    // outside of a test suite.
+
+    // NOTE: failed tests will throw a SIGTRAP in order to call the handler
+    // defined below, which will trigger a pause in any running debugger.
+    // Execution can continue normally from there. To see this in action,
+    // modify the CHECK below to fail then run the extension in a debugger
+
+    CHECK(1 + 1 == 2);
+}
+
+static void handler(const doctest::AssertData& ad) {
+    // Handles failed doctest assertions
+
+    using namespace doctest;
+
+    std::cout << Color::LightGrey << skipPathFromFilename(ad.m_file) << "(" << ad.m_line << "): ";
+    std::cout << Color::Red << failureString(ad.m_at) << ": ";
+
+    // handling only normal (comparison and unary) asserts - exceptions-related asserts have been skipped
+    if (ad.m_at & assertType::is_normal) {
+        std::cout << Color::Cyan << assertString(ad.m_at) << "( " << ad.m_expr << " ) ";
+        std::cout << Color::None << (ad.m_threw ? "THREW exception: " : "is NOT correct!\n");
+        if (ad.m_threw)
+            std::cout << ad.m_exception;
+        else
+            std::cout << "  values: " << assertString(ad.m_at) << "( " << ad.m_decomp << " )";
+    } else {
+        std::cout << Color::None << "an assert dealing with exceptions has failed!";
+    }
+
+    std::cout << std::endl;
+}
+
+namespace cesium::omniverse::tests {
+
+class CesiumOmniverseTestsPlugin final : public ICesiumOmniverseTestsInterface {
+  public:
+    void onStartup(const char* cesiumExtensionLocation) noexcept override {
+        Context::onStartup(cesiumExtensionLocation);
+    }
+
+    void onShutdown() noexcept override {
+        Context::onShutdown();
+    }
+
+    void run_all_tests(long int stage_id) noexcept override {
+
+        std::cout << "Running Cesium Omniverse Tests with stage id: " << stage_id << std::endl;
```

Comment:
Use `CESIUM_LOG_INFO` or one of the other logging macros instead of `std::cout`.

Reply:
I've just tried using CESIUM_LOG_INFO, but it turns out `onStartup` isn't actually being called so context isn't initialized, which causes the logger to segfault. It's just a matter of adding the bindings for `onStartup` (and `onShutdown`) then passing the extension location down to them, which I wasn't able to get done today.